certbot/certbot/_internal/storage.py

```diff
@@ -283,22 +282,23 @@ def _relevant(namespaces: Iterable[str], option: str) -> bool:
             any(option.startswith(namespace) for namespace in namespaces))
 
 
-def relevant_values(all_values: Mapping[str, Any]) -> Dict[str, Any]:
+def relevant_values(config: configuration.NamespaceConfig) -> Dict[str, Any]:
     """Return a new dict containing only items relevant for renewal.
 
-    :param dict all_values: The original values.
+    :param .NamespaceConfig config: parsed command line
 
     :returns: A new dictionary containing items that can be used in renewal.
     :rtype dict:
 
     """
+    all_values = config.to_dict()
     plugins = plugins_disco.PluginsRegistry.find_all()
     namespaces = [plugins_common.dest_namespace(plugin) for plugin in plugins]
 
     rv = {
         option: value
         for option, value in all_values.items()
-        if _relevant(namespaces, option) and cli.option_was_set(option, value)
```

Comment:
I am trying to think of a reason why we had this function in the first place. Did you find something in the history that made you decide to trash it? Is there actually some case where it wasn't set by the user, but it still has a non-default value? Maybe...it has a previous default value?

Reply:
yeah i struggled with this one a bit. looking at the commit that brought this in (`aaf93b65`), it seems we wanted to hold onto options in three cases:

1. it was set by the user
2. it didn't have a default (but if we're looking at its value, it must've been set by the user?)
3. it had a value different from the default (i can't think of a way this would happen except the user setting it?)

so i don't _think_ it's logically different than just calling `set_by_user()`

Reply:
What about if it was set to a non-default value during runtime?

Reply:
good call, that'd definitely result in `True` from `option_was_set`. though the current implementation of `set_by_user` does return `True` for `ArgumentSource.RUNTIME`, so that should be covered as well

Reply:
Makes sense to me. Perhaps earlier we were less confident in having been flagged as set during runtime, but should be good now either way.